provider/pkg/provider/invoke_helm_template.go

```diff
@@ -0,0 +1,155 @@
+	Repo        string `json:"repo,omitempty"`
+	UntarDir    string `json:"untar_dir,omitempty"`
+	Username    string `json:"username,omitempty"`
+	Verify      bool   `json:"verify,omitempty"`
+	Version     string `json:"version,omitempty"`
+}
+
+type HelmChartOpts struct {
+	HelmFetchOpts `json:"fetch_opts,omitempty"`
+
+	APIVersions []string               `json:"api_versions,omitempty"`
+	Chart       string                 `json:"chart,omitempty"`
+	Namespace   string                 `json:"namespace,omitempty"`
+	Path        string                 `json:"path,omitempty"`
+	ReleaseName string                 `json:"release_name,omitempty"`
+	Repo        string                 `json:"repo,omitempty"`
+	Values      map[string]interface{} `json:"values,omitempty"`
+	Version     string                 `json:"version,omitempty"`
+}
+
+// helmTemplate performs Helm fetch/pull + template operations and returns the resulting YAML manifest based on the
+// provided chart options.
+func helmTemplate(opts HelmChartOpts) (string, error) {
+	tempDir, err := ioutil.TempDir("", "helm")
+	if err != nil {
+		return "", err
+	}
+	defer os.RemoveAll(tempDir)
+
+	chart := &chart{
+		opts:     opts,
+		chartDir: tempDir,
+	}
+
+	// If Path is set, use a local Chart, otherwise fetch from a remote.
+	if len(chart.opts.Path) > 0 {
+		chart.chartDir = chart.opts.Path
+	} else {
+		err = chart.fetch()
+		if err != nil {
+			return "", err
+		}
+	}
+
+	result, err := chart.template()
+	if err != nil {
+		return "", err
+	}
+
+	return result, nil
+}
+
+type chart struct {
+	opts     HelmChartOpts
+	chartDir string
+}
+
+// fetch runs the `helm fetch` action to fetch a Chart from a remote URL.
+func (c *chart) fetch() error {
+	p := action.NewPull()
+	p.Settings = cli.New()
+	p.CaFile = c.opts.CAFile
+	p.CertFile = c.opts.CertFile
+	//p.DestDir = c.opts.Destination // TODO: Not currently used, but might be useful for caching
+	//p.Devel = c.opts.Devel // Unused
+	// c.opts.Home is unused
+	p.KeyFile = c.opts.KeyFile
+	p.Keyring = c.opts.Keyring
+	p.Password = c.opts.Password
+	// c.opts.Prov is unused
+	p.RepoURL = c.opts.HelmFetchOpts.Repo
+	p.Untar = true
+	p.UntarDir = c.chartDir
+	p.Username = c.opts.Username
+	p.Verify = c.opts.Verify
+	p.Version = c.opts.HelmFetchOpts.Version
+
+	_, err := p.Run(c.opts.Chart)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// template runs the `helm template` action to produce YAML from the Chart configuration.
+func (c *chart) template() (string, error) {
+	cfg := &action.Configuration{
+		Capabilities: chartutil.DefaultCapabilities,
+		Releases:     storage.Init(driver.NewMemory()),
+	}
+
+	if c.opts.APIVersions != nil {
+		cfg.Capabilities.APIVersions = c.opts.APIVersions
+		// TODO: add support for overriding kube version
+	}
+
+	installAction := action.NewInstall(cfg)
+	installAction.ClientOnly = true
+	installAction.DryRun = true
+	installAction.IncludeCRDs = true // TODO: handle this conditionally?
```

Comment:
We may want to consider making these user specified, at least `DryRun` and `IncludeCRDs`

Reply:
`DryRun` is the way that `helm template` is implemented in the library, so we don't need that to be user-facing.

We may expose `IncludeCRDs` later, but that would be new work, and we might be able to handle it more seamlessly in the provider so that the user doesn't have to worry about it.